tests/acceptance/lazy-routeable-engine-test.js

```diff
@@ -66,6 +63,40 @@ test('it should pause to load JS and CSS assets on deep link into a lazy Engine'
   andThen(() => verifyInitialBlogRoute(assert, this.loadEvents, this.application));
 });
 
+test('it should pause to load JS and CSS assets on deep link into a lazy Engine after retrying the transition', function(assert) {
+  assert.expect(7);
+
+  define('dummy/routes/application', () => (Ember.Route.extend({
+      actions: {
+        error: nope
+      }
+    })
+  ));
+
+  const adapterException = Ember.Test.adapter.exception;
+  const loader = this.application.__container__.lookup('service:asset-loader');
+  const jsLoader = loader.__assetLoaders.js;
+  const cssLoader = loader.__assetLoaders.css;
+  const failLoader = () => RSVP.reject('rejected');
+
+  Ember.Test.adapter.exception = nope;
+  loader.defineLoader('js', failLoader);
+  loader.defineLoader('css', failLoader);
+  visit('/routable-engine-demo/blog/new');
+
+  andThen(() => {
+    loader.defineLoader('js', jsLoader);
+    loader.defineLoader('css', cssLoader);
+    visit('/routable-engine-demo/blog/new');
+
+    andThen(() => {
+      verifyInitialBlogRoute(assert, this.loadEvents, this.application);
+      Ember.Test.adapter.exception = adapterException;
```

Comment:
This cleanup should be in an `afterEach`, that guarantees that it is properly reset if any of these tests fail...

Reply:
Ok!